profiler/profiler.go

```diff
@@ -316,7 +317,7 @@ func (r *retryer) Retry(err error) (time.Duration, bool) {
 // case of error, the goroutine will sleep and retry. Sleep duration may
 // be specified by the server. Otherwise it will be an exponentially
 // increasing value, bounded by maxBackoff.
```

Comment:
Please update the function doc since the return value list changed. A comment along the lines of "An error is returned to indicate a permanent profile creation failure in which case the caller must break the polling loop."